presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/ShardCleaner.java

```diff
@@ -352,12 +356,19 @@ synchronized void cleanLocalShards()
             }
         }
 
-        // delete files marked earlier than the clean interval
-        long threshold = ticker.read() - localCleanTime.roundTo(NANOSECONDS);
-        Set<UUID> deletions = shardsToClean.entrySet().stream()
-                .filter(entry -> entry.getValue() < threshold)
-                .map(Map.Entry::getKey)
-                .collect(toSet());
+        Set<UUID> deletions;
+        if (startup) {
```

Comment:
I'd rather create a new method to do unconditional cleanup. This method does several actions which are unnecessary at startup. By removing them you simplify the logic.